crates/wdl-engine/src/cache/lock.rs

```diff
@@ -173,3 +175,49 @@ impl Write for LockedFile {
         self.0.flush()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::time::Duration;
+
+    use tempfile::NamedTempFile;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn acquire_shared_no_create() {
+        assert!(
+            LockedFile::acquire_shared("does-not-exist", false)
+                .await
+                .unwrap()
+                .is_none(),
+            "should not file"
+        );
+    }
+
+    #[tokio::test]
+    async fn acquire_shared() {
+        let path = NamedTempFile::new().unwrap();
+        let _first = LockedFile::acquire_shared(path.path(), true)
+            .await
+            .unwrap()
+            .expect("should have locked file");
+        let _second = LockedFile::acquire_shared(path.path(), true)
+            .await
+            .unwrap()
+            .expect("should have locked file");
+    }
+
+    #[tokio::test]
+    async fn acquire_exclusive() {
+        let path = NamedTempFile::new().unwrap();
+        let _first = LockedFile::acquire_exclusive(path.path()).await.unwrap();
+
+        // As acquiring a shared lock will dead lock, wait at most 3 seconds for
+        // acquiring the shared lock and panic if we acquire it
+        tokio::select! {
+            _ = tokio::time::sleep(Duration::from_secs(3)) => {}
+            _ = LockedFile::acquire_shared(path.path(), true) => panic!("a shared lock should not be acquired")
+        }
```

Comment:
This is a particularly silly way of doing this; I have a change ready to go to simply try to acquire a shared lock using `File::try_lock_shared` and ensuring the operation returns a would-block error.

Waiting on GitHub to recover from its outage so I can push to my repository.